app/controllers/assignments_controller.rb

```diff
@@ -37,16 +37,15 @@ def create
         if assignment_form_params[:assignment][:directory_path].blank?
           assignment_form_params[:assignment][:directory_path] = "assignment_#{assignment_form_params[:assignment][:id]}"
         end
-        ques_array = assignment_form_params[:assignment_questionnaire]
-        due_array = assignment_form_params[:due_date]
-        ques_array.each do |cur_questionnaire|
-          cur_questionnaire[:assignment_id] = assignment_created.id.to_s
+
+        assignment_form_params[:assignment_questionnaire].each do |cur_questionnaire|
+          cur_questionnaire[:assignment_id] = exist_assignment.id.to_s
```

Comment:
Pass by reference?

Reply:
Yes, here is the reference from the net.
Each cur_due inside the loop is a reference to a hash within assignment_form_params[:due_date]. When you set cur_due[:parent_id] = exist_assignment.id.to_s, it updates the actual hash within assignment_form_params[:due_date], so no additional assignment is necessary.